lib/cfa/grub2/install_device.rb

```diff
@@ -14,6 +14,9 @@ def self.parse(string)
 
       # gets list of devices and create file content from it
       def self.serialize(data)
+        # do not modify original data as serialize is not end of world
+        data = data.dup
```

Comment:
Actually I do not like much shadowing the arguments by local variable, what about renaming the argument? 

```ruby
  def self.serialize(input_data)
    data = input_data.dup
    ...
```